doc/mgmt/redis_client_manager.md

```diff
@@ -0,0 +1,176 @@
+- RPC: Remote Procedure Call, used to interface with sonic-mgmt-common.
+- CVL: [Config Validation Library](https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/Management%20Framework.md#3228-config-validation-library-cvl)
+
+### Overview
+
+The translib framework that lives in sonic-mgmt-common is responsible for carrying out various config and telemetry related tasks, such as serving gNMI GETs, SETs, and SUBSCRIBEs. In order to do this, translib creates and uses many Redis Clients to read/write the Redis databases. Redis Clients support connection pooling, which consists of keeping a pool of active connections that can be used instead of creating a new one each time one is needed. Connection pools significantly reduce the number of connections made to Redis and reduce stress on the Redis TCP/Unix sockets. Since the connection pools are held and managed by a Redis Client object, these Redis Client objects must be reused instead of being closed after each use.
+
+### Requirements
+
+- Redis Clients with a connection pool must be reused wherever possible instead of creating a new Redis Client each time one is needed, within sonic-mgmt-common.
+- Transactional Redis operations must be handled correctly. Connection pools do not support transactions.
+- The number of Redis Clients and connections to Redis must be tracked for the life of the server.
+
+### Architecture Design
+
+This design does not change the existing SONiC Architecture. This design adds a new API that is meant to be a replacement for redis.NewClient and redis.Close.
+
+### High-Level Design
+
+This design introduces a new API that manages all Redis connections within sonic-mgmt-common called Redis Client Manager (RCM). With a central manager, Redis Clients can be kept open after an RPC completes, reused, and usage statistics can be tracked. This will also abstract most of the Redis logic away from the user and avoid misuse of the go-redis library. Since the RCM will be the only place in sonic-mgmt-common where new Redis Clients are created, any client/connection creation failure can be exposed here. By pinging a newly created client, UMF can identify and log any connection failures.
+
+The new API will expose four new functions that will replace [redis.NewClient](https://pkg.go.dev/github.com/go-redis/redis/v8#NewClient) and [redis.Close](https://pkg.go.dev/github.com/go-redis/redis/v8#Client.Close).
+
+#### RedisClient(db DBNum)
+
+This RedisClient function takes in a database number, defined in [db.go](https://github.com/sonic-net/sonic-mgmt-common/blob/master/translib/db/db.go#L136), and returns an existing Redis Client to the caller. This Redis Client can be shared across multiple goroutines and has a connection pool. It cannot be used for [Transactional Redis Operations](https://redis.io/docs/latest/develop/interact/transactions/) (such as MULTI, EXEC, PSUBSCRIBE, etc...).
```

Comment:
I introduced a new DB option called `TransactionsRequired` (boolean). Callers of NewDB will need to set this flag in order to get a unique Redis client in the DB that is returned. This will be handled during the refactoring.

Using `IsWriteDisabled` is another option that we can consider. However, I created this new option to be more explicit.

Adding this detail to the doc.

Reply:
Done. Replaced with `ForceNewRedisConnection`.